recipes/argon2/all/conanfile.py

```diff
@@ -0,0 +1,62 @@
+from conans import ConanFile, tools
+from conans.errors import ConanInvalidConfiguration
+import os
+
+class Argon2Conan(ConanFile):
+    name = "argon2"
+    license = "Apache 2.0", "CC0-1.0"
+    homepage = "https://github.com/P-H-C/phc-winner-argon2"
+    url = "https://github.com/conan-io/conan-center-index"
+    description = "Argon2 password hashing library"
+    topics = ("argon2", "crypto", "password hashing")
+    settings = "os", "compiler", "build_type", "arch"
+    options = {"shared": [True, False], "fPIC": [True, False]}
+    default_options = {"shared": False, "fPIC": True}
+
+    @property
+    def _arch(self):
+        return str(self.settings.arch).replace("_", "-")
```

Comment:
What architectures does it support?
Only Intel ones or also arm based ones?

Reply:
By default this is build with `-march=native` [option](https://github.com/P-H-C/phc-winner-argon2/blob/440ceb9612d5a20997e3e12728542df2de713ca4/Makefile#L51) , but compiled with doesn't work on others processors. I compiled this on amd and intel

Reply:
Oh, it's just about adding a `-march` option.
I thought this was about using intrinsic functions or assembly.

The arch is obsolete, but I don't think `-march=x86` exists.
It should be i686 instead, I guess.
See https://gcc.gnu.org/onlinedocs/gcc/x86-Options.html

Reply:
Yes, but this code changes only `_` to  `-` bacause there is no march `x86_64`  like conan arch shows in profile `arch=x86_64`

Reply:
You're right about x86_64, but it will fail when running:
`conan create . argon/20190702@ -s arch=x86 -s os=Linux`

So the recipe needs to do one of the following:
- patch out `-march=...` from the Makefile.
This is, imho, the best solution because it will work with *all* archs. Users can even specify more specialized architectures by adding `-march=knl` to `CFLAGS` in their profile.
- make this `_arch` member property return a value using a dict (such as it returns i686 for x86).
This is, imho, not a good solution because it will not work for arm/risvc/... architectures

Reply:
Yes, you are right, patch our Makefile would be the best solution.
Could you prepare it?

Reply:
https://github.com/nicraMarcin/conan-center-index/pull/2

Reply:
Great